db/mongo/mongoClient_test.go

```diff
@@ -0,0 +1,214 @@
+				Phone: "03005478963",
+			},
+			},
+			wantErr: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			m, _ := NewMongoClient(db.Option{})
+			if err := m.UpdateStudent(tt.args.student); (err != nil) != tt.wantErr {
+				t.Errorf("UpdateStudent() error = %v, wantErr %v", err, tt.wantErr)
+			}
+		})
+	}
+}
+
+func Test_mongoStd_GetStudentByID(t *testing.T) {
+	os.Setenv("DB_PORT", "27017")
+	os.Setenv("DB_HOST", "student-portal-mongo-db")
+
+	m, _ := NewMongoClient(db.Option{})
+	student := &models.Student{
+		Name:  "Ali",
+		Age:   22,
+		Level: "Bachelor",
+		Phone: "03005124896",
+	}
+	_, _ = m.AddStudent(student)
+	type args struct {
+		id string
+	}
+	tests := []struct {
+		name    string
+		args    args
+		want    *models.Student
+		wantErr bool
+	}{
+		{
+			name:    "success - get student by ID",
+			args:    args{id: student.ID},
+			want:    student,
+			wantErr: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := m.GetStudentByID(tt.args.id)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("GetStudentByID() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+			if !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("GetStudentByID() got = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func Test_mongoStd_RemoveStudentByID(t *testing.T) {
+	os.Setenv("DB_PORT", "27017")
+	os.Setenv("DB_HOST", "student-portal-mongo-db")
+
+	m, _ := NewMongoClient(db.Option{})
+	student := &models.Student{
+		Name:  "Usman",
+		Age:   22,
+		Level: "Bachelor",
+		Phone: "03005124896",
+	}
+	_, _ = m.AddStudent(student)
+	type args struct {
+		id string
+	}
+	tests := []struct {
+		name    string
+		args    args
+		want    *models.Student
+		wantErr bool
+	}{
+		{
+			name:    "success - delete student by ID",
+			args:    args{id: student.ID},
+			want:	 student,
+			wantErr: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			err := m.RemoveStudentByID(tt.args.id)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("RemoveStudentByID() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+		})
+	}
+}
+
+func Test_mongoStd_ListStudent(t *testing.T) {
+	os.Setenv("DB_PORT", "27017")
+	os.Setenv("DB_HOST", "student-portal-mongo-db")
+	m, _ := NewMongoClient(db.Option{})
+	student := []models.Student{{
+		Name:  "Shah",
+		Age:   22,
+		Level: "Bachelor",
+		Phone: "03005124896"},
+		{
+			Name:  "Ali",
+			Age:   22,
+			Level: "Bachelor",
+			Phone: "03005124896",
+		},
+	}
+	_, _ = m.AddStudent(&student[0])
+	_, _ = m.AddStudent(&student[1])
+	tests := []struct {
+		name    string
+		want    *[]models.Student
+		wantErr bool
+	}{
+		{
+			name:    "success - list students from db",
+			want:    &student,
+			wantErr: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := m.ListStudent(map[string]interface{}{"age":22,"level":"Bachelor"}, int64(10), int64(5))
```

Comment:
Parameters shouldn't be passed directly here.. they should be in tests so they can be dynamically assigned for different test cases.